src/HttpClient/Plugin/GzipEncoderPlugin.php

```diff
@@ -18,25 +19,27 @@
 final class GzipEncoderPlugin implements PluginInterface
 {
     /**
-     * @var StreamFactoryInterface The PSR-17 stream factory
+     * @var HttplugStreamFactoryInterface|StreamFactoryInterface The PSR-17 stream factory
      */
     private $streamFactory;
 
     /**
      * Constructor.
      *
-     * @param StreamFactoryInterface|null $streamFactory The stream factory
+     * @param HttplugStreamFactoryInterface|StreamFactoryInterface|null $streamFactory The stream factory
      *
      * @throws \RuntimeException If the zlib extension is not enabled
      */
-    public function __construct(?StreamFactoryInterface $streamFactory = null)
+    public function __construct($streamFactory = null)
     {
         if (!\extension_loaded('zlib')) {
             throw new \RuntimeException('The "zlib" extension must be enabled to use this plugin.');
         }
 
         if (null === $streamFactory) {
             @trigger_error(sprintf('A PSR-17 stream factory is needed as argument of the constructor of the "%s" class since version 2.1.3 and will be required in 3.0.', self::class), E_USER_DEPRECATED);
+        } elseif (!$streamFactory instanceof HttplugStreamFactoryInterface && !$streamFactory instanceof StreamFactoryInterface) {
+            throw new \InvalidArgumentException(sprintf('The $streamFactory argument must be an instance of either the "%s" or the "%s" interface.', HttplugStreamFactoryInterface::class, StreamFactoryInterface::class));
```

Comment:
I would like to throw a different message when HTTPlug is used here, to deprecate its direct usage. Is it possible?

Reply:
Yes it is, I didn't because of a few reasons:

- We still didn't decide if we want to stop supporting Httplug `1.x` in `3.0` regardless of the switch to PSR classes, although I'm strongly in favor of doing it
- Deprecating Httplug here means that it will be the only place in the entire codebase where it's discouraged to rely on the legacy Httplug factories, which sounds strange and incoherent. We can treat is as the starting point for deprecating Httplug `1.x`, but we must all agree on this
- Deprecating Httplug here means that we have to handle skipping the throwing of the error if the plugin is instantiated from Sentry itself, and to do it we have to rely on `boolean` flags like we do in other places of the codebase. It's fine because we will have to do it at some point if we decide to deprecate the support of Httplug `1.x`, but again we have to decide this before imho

Reply:
Ok I'm fine on postponing the deprecation discussion and merge this as is.